src/angular-google-analytics.js

```diff
@@ -596,12 +673,23 @@
           });
           _analyticsJs(function () {
             if (that._enhancedEcommerceEnabled()) {
-              $window.ga('ec:addPromo', {
-                id: productId,
-                name: name,
-                creative: creative,
-                position: position
-              });
+              if (angular.isArray(accountId)) {
```

Comment:
Well you're right the code is not very nice. I'll take a look, if I find another way to handle that. The challenge will be, that multiple trackers will not effect all ga command. Tracking the page views by event is a nice idea, but it will also cause problems in case of custom dimensions. From my point of view, custom dimensions will be set up in a lot of cases during the angular app get's bootstrapped. If the event causes tracking the pageview before, the dimensions will not be tracked.